source/cdk/cdk_slurm_stack.py

```diff
@@ -1096,6 +1098,58 @@ def create_lambdas(self):
     def create_parallel_cluster_lambdas(self):
         self.create_callSlurmRestApiLambda()
 
+        self.parallel_cluster_lambda_layer = aws_lambda.LayerVersion(self, "ParallelClusterLambdaLayer",
+            description = 'ParallelCluster Layer',
+            code = aws_lambda.Code.from_bucket(
+                s3.Bucket.from_bucket_name(self, 'ParallelClusterBucket', f"{self.cluster_region}-aws-parallelcluster"),
+                f"parallelcluster/{self.config['slurm']['ParallelClusterConfig']['Version']}/layers/aws-parallelcluster/lambda-layer.zip"
+            ),
+            compatible_architectures = [
+                aws_lambda.Architecture.ARM_64,
+                aws_lambda.Architecture.X86_64,
+            ],
+            compatible_runtimes = [
+                aws_lambda.Runtime.PYTHON_3_9,
+                aws_lambda.Runtime.PYTHON_3_10,
+                # aws_lambda.Runtime.PYTHON_3_11, # Doesn't work: No module named 'rpds.rpds'
+            ],
+        )
+
+        createParallelClusterLambdaAsset = s3_assets.Asset(self, "CreateParallelClusterAsset", path="resources/lambdas/CreateParallelCluster")
+        self.create_parallel_cluster_lambda = aws_lambda.Function(
+            self, "CreateParallelClusterLambda",
+            function_name=f"{self.stack_name}-CreateParallelCluster",
+            description="Create ParallelCluster from json string",
+            memory_size=2048,
+            runtime=aws_lambda.Runtime.PYTHON_3_9,
+            architecture=aws_lambda.Architecture.X86_64,
+            timeout=Duration.minutes(15),
+            log_retention=logs.RetentionDays.INFINITE,
+            handler="CreateParallelCluster.lambda_handler",
+            code=aws_lambda.Code.from_bucket(createParallelClusterLambdaAsset.bucket, createParallelClusterLambdaAsset.s3_object_key),
+            layers=[self.parallel_cluster_lambda_layer],
+            vpc = self.vpc,
+            allow_all_outbound = True
+        )
+        self.create_parallel_cluster_lambda.add_to_role_policy(
```

Comment:
The permissions are too permissive.
Tighten them up to the minimum required for the Lambda.